numba/npyufunc/dufunc.py

```diff
@@ -0,0 +1,33 @@
+from __future__ import absolute_import, print_function, division
+import numpy
+
+from numba import jit, typeof, utils
+from .. import numpy_support
+from . import _internal, ufuncbuilder
+
+class DUFunc(_internal._DUFunc):
+    def __init__(self, py_func, **kws):
+        dispatcher = jit(target='npyufunc')(py_func)
+        super(DUFunc, self).__init__(dispatcher, **kws)
+        # Clean up _DUFunc keyword arguments, and assume anything that
+        # remains is a compiler option.
+        kws.pop('identity', None)
+        kws.pop('nin', None)
+        kws.pop('nout', None)
+        self.targetoptions = kws
+
+    def _compile_for_args(self, *args, **kws):
+        assert len(kws) == 0
+        argtys = tuple(numpy_support.map_arrayscalar_type(arg)
+                       if numpy_support.is_arrayscalar(arg)
+                       else typeof(arg)
+                       for arg in args[:self.ufunc.nin])
+        if all(hasattr(argty, 'dtype') for argty in argtys):
```

Comment:
Well if you are going to wink about it, then I'll agree to the recommended change. (* smirk *)  As far as the later dtype stuff, I don't think I was thinking clearly about it, and implemented an all-or-nothing approach.  This would clearly break where I'm passing a scalar and an array, so again, I'll agree to your comment.